File: ai-job-portal-backend/candidate-service/src/main/java/com/prahlad/aijobportal/candidateservice/candidate/service/CandidateService.java
package com.prahlad.aijobportal.candidateservice.candidate.service;

import com.prahlad.aijobportal.candidateservice.candidate.dto.request.CreateCandidateProfileRequest;
import com.prahlad.aijobportal.candidateservice.candidate.dto.request.UpdateCandidateProfileRequest;
import com.prahlad.aijobportal.candidateservice.candidate.dto.response.CandidateProfileResponse;
import com.prahlad.aijobportal.candidateservice.candidate.dto.response.ProfileCompletionResponse;

import java.util.UUID;

public interface CandidateService {

    CandidateProfileResponse createProfile(UUID userId, String bearerToken, CreateCandidateProfileRequest request);

    CandidateProfileResponse getProfile(UUID userId);

    CandidateProfileResponse updateProfile(UUID userId, UpdateCandidateProfileRequest request);

    void deleteProfile(UUID userId);

    ProfileCompletionResponse getProfileCompletion(UUID userId);
}

File: ai-job-portal-frontend/src/features/profile/services/candidate.service.ts
import { apiClient } from "@/services/api-client";
import type { ApiResponse } from "@/types/api";
import type {
  CandidateProfileResponse,
  CreateCandidateProfileRequest,
  ProfileCompletionResponse,
  UpdateCandidateProfileRequest,
} from "@/features/profile/types";

/** Maps 1:1 to candidate-service CandidateController (/candidate/profile). */
export const candidateService = {
  getProfile: () =>
    apiClient.get<ApiResponse<CandidateProfileResponse>>("/candidate/profile").then((res) => res.data),

  createProfile: (payload: CreateCandidateProfileRequest) =>
    apiClient
      .post<ApiResponse<CandidateProfileResponse>>("/candidate/profile", payload)
      .then((res) => res.data),

  updateProfile: (payload: UpdateCandidateProfileRequest) =>
    apiClient
      .put<ApiResponse<CandidateProfileResponse>>("/candidate/profile", payload)
      .then((res) => res.data),

  deleteProfile: () => apiClient.delete<ApiResponse<null>>("/candidate/profile").then((res) => res.data),

  getProfileCompletion: () =>
    apiClient
      .get<ApiResponse<ProfileCompletionResponse>>("/candidate/profile/completion")
      .then((res) => res.data),
};
